extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -26,12 +27,19 @@
         return new Supplier<MongoClientSupport>() {
             @Override
             public MongoClientSupport get() {
+
                 List<ConnectionPoolListener> connectionPoolListeners = new ArrayList<>(connectionPoolListenerSuppliers.size());
                 for (Supplier<ConnectionPoolListener> item : connectionPoolListenerSuppliers) {
                     connectionPoolListeners.add(item.get());
                 }
+                final CommandListener commandListener = Arc.container().instance(CommandListener.class).get();
 
-                return new MongoClientSupport(codecProviders, bsonDiscriminators, connectionPoolListeners, disableSslSupport);
+                MongoClientSupport mongoClientSupport = new MongoClientSupport(codecProviders, bsonDiscriminators,
+                        connectionPoolListeners, disableSslSupport);
+                if (null != commandListener) {
+                    mongoClientSupport.addCommandListener(commandListener);
+                }
```

Comment:
This should not be needed, just pass the Command listener list as we do for all other MongoDB related classes